numpy/core/tests/test_api.py

```diff
@@ -50,6 +50,7 @@ def test_array_array():
     assert_equal(np.array("1", dtype=S5), np.ones((), dtype=S5))
 
     # test unicode
+    # TODO[gh-15306]: can this be deleted?
     _unicode = globals().get("unicode")
     if _unicode:
```

Comment:
I wasn't sure on this, can I delete this now? do I need to wait till np.unicode is deprecated?

Reply:
Yes, this can be removed immediately, because its referring to the builtin unicode which does not exist in python 3.

`np.unicode` has always been useless, it's just another name for `str`. `np.unicode_` is probably going to be around for much longer ,but that's not relevant here.

Having said that, I think what we should actually do is:
* Change the test above this line to use `bytes` instead of `str`, and `b"..."` instead of `"..."` (but leave `S2` etc alone)
* Change the test below this line to use `str` instead of `_unicode`, and `"..."` instead of `_unicode("...")` (but leave `U2` etc alone)

Reply:
I assume you want to leave this to a follow-up?

Reply:
I think that's a better test; not sure if I did exactly what you expected.

Reply:
I'm not sure you you mean by that. I expected you to either change this code, or comment saying you'd rather not in this PR. Did you forget to push?

Reply:
Ugh, it never pushed it was waiting for my passphrase.